pinot-core/src/main/java/org/apache/pinot/core/util/QueryOptionsUtils.java

```diff
@@ -45,6 +47,11 @@ public static Long getTimeoutMs(Map<String, String> queryOptions) {
     }
   }
 
+  public static boolean isAndScanReorderingEnabled(Map<String, String> queryOptions) {
+    return !MapUtils.isEmpty(queryOptions)
+        && queryOptions.containsKey(CommonConstants.Query.OptimizationSwitches.AND_SCAN_REORDERING);
```

Comment:
Don't do the null check here to be consistent. You can check it on the caller side if necessary.
```suggestion
    return Boolean.parseBoolean(queryOptions.get(AND_SCAN_REORDERING));
```

Reply:
Done